plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -413,12 +448,21 @@ func (p *pusher) convertEvent(e logs.LogEvent) *cloudwatchlogs.InputLogEvent {
 			// a valid timestamp and use the last valid timestamp for new entries that does
 			// not have a timestamp.
 			t = p.lastValidTime
+            if p.lastUpdateTime != 0 {
+              // Check when timestamp has an interval of 5 days.
+              // 432000 seconds = 5 days
+              // p.lastUpdateTime -> is set in miliseconds
+                if ((time.Now().UnixNano() / 1000000000) - (p.lastUpdateTime / 1000)) > 432000 {
```

Comment:
>100000000 
is this to convert from Nano seconds to seconds? can we not just use Unix() instead of UnixNano(), is there any particular reason we are using UnixNano ?

Reply:
Agreed. The log timestamps that CWL accepts do not accept nanosecond precision. There's no need for us to use that level of precision.

Reply:
Better yet, store the `lastUpdateTime` as a `time.Time` so you can do the check without all this conversion.

```go
time.Since(p.lastUpdateTime) > [name_of_threshold_variable]
```

where [name_of_threshold_variable] is like `5 * 24 * time.Hour`

Reply:
Fixed this, to follow format as Jeffrey suggested
```
			t = p.lastValidTime
			if !p.lastUpdateTime.IsZero() {
				// Check when timestamp has an interval of 5 days.
				if time.Since(p.lastUpdateTime) > elaspedPeriod {
					p.Log.Warnf("Unable to parse timestamp, using last valid timestamp %v: which is older than 5 days for log group %v: ", p.lastValidTime, p.Group)
				}
			}
```